File: src/Software Engineer Prepare Kit/15/solution.ts
export default function findTaskPairForSlot(taskDurations: number[], slotLength: number): number[] {
	const seenTasks = new Map<number, number>()

	for (let i = 0; i < taskDurations.length; i++) {
		const currentDuration = taskDurations[i]!
		const neededComplement = slotLength - currentDuration

		if (seenTasks.has(neededComplement)) {
			return [seenTasks.get(neededComplement)!, i]
		}

		seenTasks.set(currentDuration, i)
	}

	return [-1, -1]
}

File: src/Software Engineer Prepare Kit/15/solution.go
package main

func FindTaskPairForSlot(taskDurations []int32, slotLength int32) []int32 {
	seenTasks := make(map[int32]int32)

	for i, duration := range taskDurations {
		complement := slotLength - duration

		if _, hasComplement := seenTasks[complement]; hasComplement {
			return []int32{seenTasks[complement], int32(i)}
		}

		seenTasks[duration] = int32(i)
	}

	return []int32{-1, -1}
}
